docs/book/src/tasks/experimental-features/cluster-classes.md

```diff
@@ -0,0 +1,341 @@
+
+To make sure our workload clusters come up with a functioning network we can define a Kindnet ConfigMap with a Kindnet ClusterResourceSet. Kindnet only offers networking for Clusters built with Kind and CAPD. This can be substituted for any other networking solution for Kubernetes e.g. Calico as used in the Quickstart guide.
+`
+The kindnet configuration file can be found in the [CAPI repo](https://raw.githubusercontent.com/kubernetes-sigs/cluster-api/main/test/infrastructure/docker/examples/kindnet-clusterresourceset.yaml).
+
+To create the resources run:
+```bash
+kubectl apply -f https://raw.githubusercontent.com/kubernetes-sigs/cluster-api/main/test/infrastructure/docker/examples/kindnet-clusterresourceset.yaml 
+```
+
+#### Apply the workload cluster
+
+The below is a Cluster definition that leverages the ClusterClass created above to define its shape.
+
+```yaml
+apiVersion: cluster.x-k8s.io/v1beta1
+kind: Cluster
+metadata:
+  name: "clusterclass-quickstart"
+  namespace: default
+  labels:
+    cni: kindnet
+spec:
+  clusterNetwork:
+    services:
+      cidrBlocks: ["10.128.0.0/12"]
+    pods:
+      cidrBlocks: ["192.168.0.0/16"]
+    serviceDomain: "cluster.local"
+  topology:
+    class: clusterclass
+    version: v1.21.2
+    controlPlane:
+      replicas: 1 
+    workers:
+      machineDeployments:
+        - class: linux-worker
+          name: linux-workers
+          replicas: 1 
+```
+
+We can create the Cluster object from the file in [the CAPI repo](https://raw.githubusercontent.com/kubernetes-sigs/cluster-api/main/test/infrastructure/docker/examples/clusterclass-quickstart.yaml) with:
+
+```bash
+kubectl apply -f https://raw.githubusercontent.com/kubernetes-sigs/cluster-api/main/test/infrastructure/docker/examples/clusterclass-quickstart.yaml
+```
+
+#### Verify the workload cluster is running
+
+The cluster will now start provisioning. You can check status with:
+
+```bash
+kubectl get cluster
+```
+
+You can get a view of the cluster and its resources as they are created by running:
+
+```bash
+clusterctl describe cluster clusterclass-quickstart
+```
+
+To verify the control plane is up:
+
+```bash
+kubectl get kubeadmcontrolplane
+```
+
+You should see an output similar to the below:
+
+```bash
+NAME                            INITIALIZED   API SERVER AVAILABLE   VERSION   REPLICAS   READY   UPDATED   UNAVAILABLE
+capi-quickstart                 true                                 v1.21.2   1                  1         1
+```
+
+
+## Upgrade a Cluster using Managed Topology
+
+The `Spec.Topology` field added to the Cluster object as part of ClusterClass allows changes made on the Cluster to be propogated across all relevant objects. This turns a Kubernetes cluster upgrade into a one-touch operation.
+Looking at the cluster we just created we can see the version of the control plane and the other machines is v1.21.2.
+
+```bash
+> kubectl get kubeadmcontrolplane,machinedeployments
+
+
```

Comment:
```suggestion
```
nit: Maybe drop those empty lines. It's a lot of empty space in the rendered output.